plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DefaultJdbcMetadata.java

```diff
@@ -496,7 +496,7 @@ private static PreparedQuery asPreparedQuery(JdbcTableHandle tableHandle)
                 handle.getOtherReferencedTables(),
                 handle.getNextSyntheticColumnId());
 
-        return Optional.of(new TopNApplicationResult<>(sortedTableHandle, jdbcClient.isTopNLimitGuaranteed(session), false));
+        return Optional.of(new TopNApplicationResult<>(sortedTableHandle, jdbcClient.isTopNGuaranteed(session), false));
```

Comment:
From @raunaqmorarka's comment inhttps://github.com/trinodb/trino/pull/8171#discussion_r648071027 I felt that people might return `true` here in cases where just ordering is guaranteed but cardinality is not.

In such case the plan will not have a final TopN leading to query output having more rows than requested? Do we have a test to verify such cases?

Or maybe I misunderstood the original comment.

Reply:
> I felt that people might return true here in cases where just ordering is guaranteed but cardinality is not.

That wouldn't be correct.

We can debate the intent of the method's author, but the point is, when this returns true, the both cardinality and order must be guaranteed:

https://github.com/trinodb/trino/blob/6528d39568672d0334141a77906cd84aebe01cd9/core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PushTopNIntoTableScan.java#L105-L119

i am only trying to align method's semantics with its name

Reply:
We do have some tests which will catch cases where cardinality doesn't match expected - so my comment is a non-issue. Resolved.